web/src/read_receipts.ts

```diff
@@ -20,86 +20,101 @@ const read_receipts_api_response_schema = z.object({
-                void channel.get({
-                    url: `/json/messages/${message_id}/read_receipts`,
-                    success(raw_data) {
-                        const $modal = $("#read_receipts_modal").filter(
-                            "[data-message-id=" + message_id + "]",
-                        );
-                        // If the read receipts modal for the selected message ID is closed
-                        // by the time we receive the response, return immediately.
-                        if (!$modal.length) {
-                            return;
-                        }
-                        const data = read_receipts_api_response_schema.parse(raw_data);
-                        const users = data.user_ids.map((id) => {
-                            const user = people.get_user_by_id_assert_valid(id);
-                            return user;
-                        });
-                        users.sort(people.compare_by_name);
-
-                        const context = {
-                            users: users.map((user) => ({
-                                user_id: user.user_id,
-                                full_name: user.full_name,
-                                avatar_url: people.small_avatar_url_for_person(user),
-                            })),
-                        };
-
-                        loading.destroy_indicator($("#read_receipts_modal .loading_indicator"));
-                        if (users.length === 0) {
-                            $("#read_receipts_modal .read_receipts_info").text(
-                                $t({defaultMessage: "No one has read this message yet."}),
-                            );
-                        } else {
-                            $("#read_receipts_modal .read_receipts_info").html(
-                                $t_html(
-                                    {
-                                        defaultMessage:
-                                            "{num_of_people, plural, one {This message has been <z-link>read</z-link> by {num_of_people} person:} other {This message has been <z-link>read</z-link> by {num_of_people} people:}}",
-                                    },
-                                    {
-                                        num_of_people: users.length,
-                                        "z-link": (content_html) =>
-                                            `<a href="/help/read-receipts" target="_blank" rel="noopener noreferrer">${content_html.join(
-                                                "",
-                                            )}</a>`,
-                                    },
-                                ),
-                            );
-                            $("#read_receipts_modal .modal__container").addClass(
-                                "showing_read_receipts_list",
+            function fetch_read_receipts(): void {
```

Comment:
The indentation here is getting ridiculous. Can this be extracted to a top-level function in a prep commit with no functional changes and then called as needed? 

Check out our [GitHub guide](https://zulip.readthedocs.io/en/latest/git-guide.html) and [commit guidelines](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html) for more details.

Reply:
Hey @timabbott , I have made a pre-commit [`2ab0beab9cfc55745109d49385c06b75d9f426cf`](https://github.com/zulip/zulip/pull/32654/commits/2ab0beab9cfc55745109d49385c06b75d9f426cf) for this.